### ISSUE TITLE
VDR: channel switch fails sometimes

### ISSUE THREAD

**Issue by srehm** [catalinii/minisatip#319]
I have a VDR with satip plugin as a client to minisatip. Both minisatip and the VDR satip plugin are the most recent git versions as of today. Usually, it works like a charm, but sporadically I get just a black screen when switching to a new channel. Only solution is then two repeat the channel switch (e.g. switching one channel back and forth again).

I have cranked up the log levels on both VDR and minisatip and attached the resulting log of the last ~30 sec before such a failed channel switch. Note that at ~18:02:39 minisatip starts complaining that "no data has been sent for more than 1s". Correspondingly the VDR complains about empty RTP pakets.

I am not sure whether this is really related to minisatip. All I can say is that another VDR that has it`s own DVB card but otherwise the same software versions running (except the satip plugin of course), does not have these channel switching problems. Could still be an issue with the DVB card (a DD Cine Max S8) in the minisatip server though. Any help on that issue would be highly appreciated.
[logs.zip](https://github.com/catalinii/minisatip/files/418466/logs.zip)

**Comment by catalinii** [comment]
Hi,

I took a look to the logs, my finding so far:

All it seems is good until 18:02:38 when I see this:
[15/08 18:02:38.036 main]: tuning to 11347000(1597000) pol: v (1) sr:22000000 fec:23 delsys:dvbs2 mod:8psk rolloff:0.35 pilot: , ts clear=570383, ts pol=570383
then the filters are added:
[15/08 18:02:38.036 main]: setting filter on PID 6710 for fd 12
[15/08 18:02:38.036 main]: setting filter on PID 6720 for fd 13
[15/08 18:02:38.036 main]: setting filter on PID 6721 for fd 14
[15/08 18:02:38.036 main]: setting filter on PID 6722 for fd 15
[15/08 18:02:38.036 main]: setting filter on PID 6723 for fd 16
[15/08 18:02:38.046 main]: setting filter on PID 6731 for fd 17
[15/08 18:02:39.019 main]: setting filter on PID 18 for fd 18
 Then 200ms later you can see this,
[15/08 18:02:39.274 main]: pid 6710, fd 12, type 0 **packets 0**, d/c errs 0/0, flags 1, key 255, sids: 0 -1 -1 -1 -1 -1 -1 -1
[15/08 18:02:39.274 main]: pid 6720, fd 13, type 0 packets 0, d/c errs 0/0, flags 1, key 255, sids: 0 -1 -1 -1 -1 -1 -1 -1
[15/08 18:02:39.274 main]: pid 6721, fd 14, type 0 packets 0, d/c errs 0/0, flags 1, key 255, sids: 0 -1 -1 -1 -1 -1 -1 -1
[15/08 18:02:39.274 main]: pid 6722, fd 15, type 0 packets 0, d/c errs 0/0, flags 1, key 255, sids: 0 -1 -1 -1 -1 -1 -1 -1
[15/08 18:02:39.274 main]: pid 6723, fd 16, type 0 packets 0, d/c errs 0/0, flags 1, key 255, sids: 0 -1 -1 -1 -1 -1 -1 -1
[15/08 18:02:39.274 main]: pid 6731, fd 17, type 0 packets 0, d/c errs 0/0, flags 1, key 255, sids: 0 -1 -1 -1 -1 -1 -1 -1
[15/08 18:02:39.274 main]: pid 18, fd 18, type 0 packets 0, d/c errs 0/0, flags 1, key 255, sids: 0 -1 -1 -1 -1 -1 -1 -1

I am thinking you are on Astra 19 and those are HD streams, there should be more than 0.

Later on, I see that the card has signal:
[15/08 18:02:42.097 AD0]: no data sent for more than 1s sid: 0 for 192.168.0.104:40298
then the first TS packet is being sent...
[15/08 18:02:55.537 AD0]: Start streaming for stream 0, len 188 to handle 8 => 192.168.0.104:40298

I am not saying there is nothing on the minisatip side, but is really unlikely (you can get a log with 8 -l's ) and we can see if something is coming from the kernel, I think it's unlikely... 
The simplest way it would be to test with another card if it happens the same.

Thanks

**Comment by catalinii** [comment]
Hi, 
more information, it seems transponder related and the ones that are losing TS packets are QAM/256. 64/QAM works good.
Just as background information, the QAM/256 transponders are working with the windows driver.

I will test the new commit and let you know...

**Comment by srehm** [comment]
Hi,

yes, you are correct: the failed channel in the log was a HD channel on ASTRA 19.2E.
I have run a test with 8 -l`s and the result is the following:

When it works, I get something like

[27/08 14:12:06.074 AD0]: start writev handle 24, iiov 8
[27/08 14:12:06.074 AD0]: writev returned 1328 handle 24, iiov 8 (took 0 ms)
[27/08 14:12:06.074 AD0]: send_rtp: sent 1316 bytes for stream 0, handle 24 seq 1305 => 192.168.0.104:43605

in more or less regular intervals.

When it fails then the ouput of that segment changes to

[27/08 14:05:44.425 AD0]: start writev handle 16, iiov 1
[27/08 14:05:44.425 AD0]: writev returned 12 handle 16, iiov 1 (took 0 ms)
[27/08 14:05:44.425 AD0]: send_rtp: sent 0 bytes for stream 0, handle 16 seq 1847 => 192.168.0.104:43605
[27/08 14:05:44.425 AD0]: describe_adapter: sid 0, aid 0 => ver=1.0;src=1;tuner=1,254,1,13,11582,h,8psk,,0.35,dvbs2,22000,23;pids=5201,5202,5203,5205,5206,0,16,17,18,20
[27/08 14:05:44.425 AD0]: send_rtcp: sent 176 bytes for stream 0, handle 16 seq 1847 => 192.168.0.104:43605

Does that meas that there is no data from the card?

**Comment by catalinii** [comment]
Hi,

You ahould look for process_dmx and read_dmx lines. If there are not any it means no data is received from the card.

**Comment by srehm** [comment]
Ok, did a

minisatip -f -llllllll | grep dmx

and there are no more process_dmx or read_dmx lines in case of a failed channel switch.

**Comment by catalinii** [comment]
Send me the log to look at it and see if everything else is OK.

**Comment by srehm** [comment]
Here is the full log:
[satip.log.zip](https://github.com/catalinii/minisatip/files/441032/satip.log.zip)

**Comment by catalinii** [comment]
Hi,

It is very clear that nothign comes from the DVB card and there is signal:
[28/08 09:28:55.828 signal]: get_signal_new took 13 ms for adapter 0 handle 7 (status: 31, ber: 0, strength:254, snr: 248, max_strength: 63017, max_snr: 1585 0)

before the tune:
[28/08 09:28:52.349 AD0]: read_dmx send=1, flush_all=0, cnt 1 called for adapter 0 -> 25004 out of 25004 bytes read, 23 ms ago
[28/08 09:28:52.350 AD0]: read_dmx send=0, flush_all=0, cnt 0 called for adapter 0 -> 21432 out of 25004 bytes read, 1 ms ago

after:
[28/08 09:28:55.183 AD0]: read_dmx send=1, flush_all=1, cnt 0 called for adapter 0 -> 188 out of 25004 bytes read, 2834 ms ago
[28/08 09:28:55.753 AD0]: read_dmx send=1, flush_all=1, cnt 0 called for adapter 0 -> 188 out of 25004 bytes read, 570 ms ago
[28/08 09:28:57.724 AD0]: read_dmx send=1, flush_all=1, cnt 0 called for adapter 0 -> 188 out of 25004 bytes read, 1971 ms ago
[28/08 09:28:58.959 AD0]: read_dmx send=1, flush_all=1, cnt 0 called for adapter 0 -> 188 out of 25004 bytes read, 1235 ms ago
[28/08 09:28:59.054 AD0]: read_dmx send=1, flush_all=1, cnt 0 called for adapter 0 -> 188 out of 25004 bytes read, 95 ms ago

So basically there are some TS packets coming in... but just from time to time.

I would speak with the support/driver maintainer....

Thanks

**Comment by srehm** [comment]
Ok. I will do that. Thank you for narrowing down the problem.
